File: chapter2/toolbar_frame1.rb
require 'java'

java_import java.awt.BorderLayout
java_import java.awt.Button
java_import java.awt.FlowLayout
java_import java.awt.Frame
java_import java.awt.Panel
# Events
java_import java.awt.event.ActionListener
java_import java.awt.event.WindowAdapter
java_import java.lang.System

class ToolbarFrame1 < Frame

  def initialize()
    super("Toolbar Example (AWT)")
    size = 450, 250
    add_window_listener(WindowCloser.new)

    toolbar = Panel.new
    toolbar.layout = FlowLayout.new(FlowLayout::LEFT)
    cut = Button.new("Cut")
    cut.add_action_listener(PrintListener.new)
    toolbar.add(cut)

    copy = Button.new("Copy")
    copy.add_action_listener(PrintListener.new)
    toolbar.add(copy)

    paste = Button.new("Paste")
    paste.add_action_listener(PrintListener.new)
    toolbar.add(paste)

    add(toolbar, BorderLayout::NORTH)
  end

  class WindowCloser < WindowAdapter
    def window_closing
      System.exit()
    end
  end

  class PrintListener
    include ActionListener
    def action_performed(event)
      System.out.println(event.action_command)
    end
  end

end

tf1 = ToolbarFrame1.new
tf1.visible = true

File: chapter2/ToolbarFrame1.java
// ToolbarFrame1.java
// A simple frame containing a "toolbar made up of several java.awt.Button
// objects. We'll be converting the Buttons to JButtons in the ToolbarFrame2.java
// file.
//
import java.awt.*;
import java.awt.event.*;

public class ToolbarFrame1 extends Frame {

  Button cutButton, copyButton, pasteButton;
  public ToolbarFrame1() {
    super("Toolbar Example (AWT)");
    setSize(450, 250);
    addWindowListener(new WindowAdapter(){
      public void windowClosing(WindowEvent e) {
        System.exit(0);
      }
    });

    ActionListener printListener = new ActionListener(){
      public void actionPerformed(ActionEvent ae) {
        System.out.println(ae.getActionCommand());
      }
    };

    Panel toolbar = new Panel();
    toolbar.setLayout(new FlowLayout(FlowLayout.LEFT));

    cutButton = new Button("Cut");
    cutButton.addActionListener(printListener);
    toolbar.add(cutButton);

    copyButton = new Button("Copy");
    copyButton.addActionListener(printListener);
    toolbar.add(copyButton);

    pasteButton = new Button("Paste");
    pasteButton.addActionListener(printListener);
    toolbar.add(pasteButton);

    // The "preferred" BorderLayout add call
    add(toolbar, BorderLayout.NORTH);
  }

  public static void main(String args[]){
    ToolbarFrame1 tf1 = new ToolbarFrame1();
    tf1.setVisible(true);
  }

}
